src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py

```diff
@@ -5651,6 +5699,19 @@ def set_up_network_profile(self, mc: ManagedCluster) -> ManagedCluster:
 
         network_dataplane = self.context.get_network_dataplane()
 
+        acns = self.models.AdvancedNetworking()
+        (acns_enabled, acns_observability, acns_security) = self.context.get_acns_enablement()
+        if acns_enabled is not None:
+            acns.enabled = acns_enabled
```

Comment:
```suggestion
        (acns_enabled, acns_observability, acns_security) = self.context.get_acns_enablement()
        if acns_enabled is not None:
            acns = self.models.AdvancedNetworking(enabled=acns_enabled)
```

Reply:
> >       acns = self.models.AdvancedNetworking()
> E       AttributeError: 'AKSManagedClusterModels' object has no attribute 'AdvancedNetworking'

official azure-cli also supports some profiles which use fixed old API versions

Reply:
same for the update code path

Reply:
I see. I have updated the code paths now.